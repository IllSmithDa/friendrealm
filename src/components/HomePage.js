import React, { Component } from 'react';
import Navbar from './Navbar';
import AllVideos from './AllVideos';
import '../CSS/PageLayout.css';
import '../CSS/App.css';

export default class HomePage extends Component {
  render() {
    return (
      <div>
        <Navbar />
        <div className = 'Page-Container'>
        <h1 className="app-title-item video-title"> <b> Welcome to  Videorealm</b> </h1>
        <h4><b>Your next social media app!</b><br/><br/></h4>
        <h4 className='app-title-item'><br />Latest Videos </h4>
          <AllVideos/>
        </div>
      </div>
    );
  }
}