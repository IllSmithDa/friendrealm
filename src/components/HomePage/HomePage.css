.homepage-container {
  width: 80%;
  margin: auto;
  padding-bottom: 2rem;
}
.app-title-container {
  height: 250px;
  margin-top: 2em;
}
.title-container {
  display:flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  height: 250px;
  border-radius: 1rem;
}
.title-container h1 {
  font-size: 2.5em;
  font-weight: 800;
  color: #FFF;
}
.videos-container {
  padding: 1rem;
  margin-top: 2rem;
  background-color:  #2E4053;
  border-radius: 1rem;
}
.videos-container h1 {
  color: #FFF;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 2rem;

}
#title-slide {
  background-image: linear-gradient(180deg, #111111, rgba(0, 100, 200, 0.07) 102%),
  url('https://images.pexels.com/photos/838990/pexels-photo-838990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}
#first-news {
  background-image: linear-gradient(84deg, #000000, rgba(0, 0, 0, 0.07) 102%), 
  url('https://images.pexels.com/photos/2246476/pexels-photo-2246476.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}
#second-news {
  background-image: linear-gradient(180deg, #000000, rgba(200, 200, 200, 0.07) 102%),
  url('https://images.pexels.com/photos/36487/above-adventure-aerial-air.jpg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260');
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}


@media(max-width: 850px) {
  .title-container h1 {
    font-size: 1.3rem;
  }
  .homepage-container {
    width: 100%;
    margin: auto;
  }
  .title-container {
    border-radius: 0;
  }
  .videos-container h1  {
    font-size: 1.5rem;
  }
}