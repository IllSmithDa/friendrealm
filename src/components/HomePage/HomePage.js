/* eslint-disable import/no-unresolved */
import React from 'react';
import Navbar from '../Navbar/Navbar';
import Footer from '../Footer/Footer';
import AllVideos from '../AllVIdeos/AllVideos';
import PopularVideos from '../PopularVideos/PopularVideos';
import './HomePage.css';

export default function HomePage() {
  return (
    <div>
      <Navbar />
      <div className="homepage-container">
        <div id="demo" className="app-title-container carousel slide" data-ride="carousel">
          <ul className="carousel-indicators">
            <li data-target="#demo" data-slide-to="0" className="active" />
            <li data-target="#demo" data-slide-to="1" />
            <li data-target="#demo" data-slide-to="2" />
          </ul>
          <div className="carousel-inner">
            <div className="carousel-item active">
              <div className="title-container" id="title-slide">
                <h1>Videorealm</h1>
              </div>
            </div>
            <div className="carousel-item">
              <div className="title-container" id="first-news">
                <h1>Checkout New Videos</h1>
              </div>
            </div>
            <div className="carousel-item">
              <div className="title-container" id="second-news">
                <h1>Sign Up for Free</h1>
              </div>
            </div>
          </div>
          <a className="carousel-control-prev" href="#demo" data-slide="prev">
            <span className="carousel-control-prev-icon" />
          </a>
          <a className="carousel-control-next" href="#demo" data-slide="next">
            <span className="carousel-control-next-icon" />
          </a>
        </div>
        <div className="videos-container">
          <h1>Popular Videos </h1>
          <PopularVideos />
        </div>
        <div className="videos-container">
          <h1>Latest Videos </h1>
          <AllVideos />
        </div>
      </div>
      <Footer />
    </div>
  );
}
