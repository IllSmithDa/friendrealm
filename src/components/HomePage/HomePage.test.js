import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';

test('Should match snapshot', () => {
  const component = renderer.create(
    <HomePage />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.getQueryResponse();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
