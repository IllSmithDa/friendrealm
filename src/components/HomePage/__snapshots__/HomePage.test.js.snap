// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div>
  <div
    className="navbar-container"
  >
    <div
      className="navbar-grid"
    >
      <div>
        <div
          className="navbar-home-container"
        >
          <img
            alt="home-page"
            className="navbar-icon-home"
            onClick={[Function]}
            src="https://img.icons8.com/dusk/64/000000/home-page.png"
          />
        </div>
        <div
          className="mobile-button-container"
        >
          <div
            className="btn-group navbar-mobile-container "
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              className="mobile-nav-button"
              data-toggle="dropdown"
              type="button"
            >
              ☰
            </button>
            <div
              className="dropdown-menu dropdown-container"
            >
              <a
                className="dropdown-item"
                href="/"
              >
                HOME
              </a>
              <span
                className="dropdown-item"
                id="login"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                LOGIN
              </span>
              <span
                className="dropdown-item"
                id="create-account"
                onClick={[Function]}
                role="button"
                tabIndex={-1}
              >
                SIGN UP FOR FREE
              </span>
            </div>
          </div>
          <span>
            ☰
          </span>
        </div>
      </div>
      <div>
        <div
          className="searchbar-grid"
        >
          <div
            className="searchbar-container"
          >
            <input
              id="search-field"
              maxLength="350"
              onChange={[Function]}
              placeholder="Search for videos"
              type="text"
            />
            <img
              alt="search-video"
              className="navbar-icon-item"
              onClick={[Function]}
              src="https://img.icons8.com/doodle/48/000000/search--v1.png"
            />
          </div>
        </div>
      </div>
      <div
        className="account-container"
      >
        <div>
          <span
            id="login"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          >
            LOGIN
          </span>
          <span
            id="create-account"
            onClick={[Function]}
            role="button"
            tabIndex={-1}
          >
            SIGN UP FOR FREE
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="homepage-container"
  >
    <div
      className="app-title-container carousel slide"
      data-ride="carousel"
      id="demo"
    >
      <ul
        className="carousel-indicators"
      >
        <li
          className="active"
          data-slide-to="0"
          data-target="#demo"
        />
        <li
          data-slide-to="1"
          data-target="#demo"
        />
        <li
          data-slide-to="2"
          data-target="#demo"
        />
      </ul>
      <div
        className="carousel-inner"
      >
        <div
          className="carousel-item active"
        >
          <div
            className="title-container"
            id="title-slide"
          >
            <h1>
              Videorealm
            </h1>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <div
            className="title-container"
            id="first-news"
          >
            <h1>
              Checkout New Videos
            </h1>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <div
            className="title-container"
            id="second-news"
          >
            <h1>
              Sign Up for Free
            </h1>
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        data-slide="prev"
        href="#demo"
      >
        <span
          className="carousel-control-prev-icon"
        />
      </a>
      <a
        className="carousel-control-next"
        data-slide="next"
        href="#demo"
      >
        <span
          className="carousel-control-next-icon"
        />
      </a>
    </div>
    <div
      className="videos-container"
    >
      <h1>
        Popular Videos 
      </h1>
      <div>
        <div
          className="pop-video-container"
        />
        <div
          className="more-pop-videos"
        >
          <span
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
             See More videos 
          </span>
        </div>
      </div>
    </div>
    <div
      className="videos-container"
    >
      <h1>
        Latest Videos 
      </h1>
      <div>
        <div
          className="all-videos-container"
        />
        <div
          className="more-all-videos"
        >
          <span
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
             See More videos 
          </span>
        </div>
      </div>
    </div>
  </div>
  <br />
  <br />
  <br />
  <div
    className="footer-container"
  >
    <div
      className="footer-grid"
    >
      <div
        className="footer-section"
      >
        <h1>
          About
        </h1>
        <ul>
          <li>
            <a
              href="/wonderlandcustom/s/"
            >
              Infinte
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/news"
            >
              News
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/support"
            >
              Help
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Company
            </a>
          </li>
        </ul>
      </div>
      <div
        className="footer-section"
      >
        <h1>
          Products
        </h1>
        <ul>
          <li>
            <a
              href="/wonderlandcustom/s/store"
            >
              Electronics
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/store"
            >
              Software
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/store"
            >
              Subscriptions
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/store"
            >
              Other Products
            </a>
          </li>
        </ul>
      </div>
      <div
        className="footer-section"
      >
        <h1>
          Support
        </h1>
        <ul>
          <li>
            <a
              href="/wonderlandcustom/s/support"
            >
              Product support
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Forums
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Order Status
            </a>
          </li>
          <li>
            <a
              href="/wonderlandcustom/s/contact-us"
            >
              Contact Us
            </a>
          </li>
        </ul>
      </div>
      <div
        className="footer-section"
      >
        <h1>
          Legal
        </h1>
        <ul>
          <li>
            <a
              href="/"
            >
              Privacy Policy
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Terms of Service
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              End User License Agreements
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Legal Information
            </a>
          </li>
          <li>
            <a
              href="/"
            >
              Cookie Policy
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="trademark-set"
    >
      <p>
        © 2020 Infinite Softworks Co. All Rights Reserved. Infinite Softworks co. logo are trademarks of Infinite Softworks co. in the U.S. and/or other countries.
      </p>
      <a
        href="https://icons8.com/icons/set/low-volume"
        target="_blank"
      >
        Low Volume icon
      </a>
       icon by 
      <a
        href="https://icons8.com"
        target="_blank"
      >
        Icons8
      </a>
    </div>
  </div>
</div>
`;
